tests/pods_test.go

```diff
@@ -1,16 +1,113 @@
+const defaultNumberOfReplicas = 2
+
 var _ = Describe("Pods", func() {
-	Context("Check the client", func() {
-		It("should not fail", func() {
-			_, err := testClient.KubeClient.CoreV1().Pods("").List(v1.ListOptions{})
+	Context("Check the pod mutating webhook", func() {
+		AfterEach(func() {
+			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
+				podList := &corev1.PodList{}
+				err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{Namespace: namespace}, podList)
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, podObject := range podList.Items {
+					err = testClient.VirtClient.Delete(context.TODO(), &podObject)
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				Eventually(func() int {
+					podList := &corev1.PodList{}
+					err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{Namespace: namespace}, podList)
+					Expect(err).ToNot(HaveOccurred())
+					return len(podList.Items)
+
+				}, timeout, pollingInterval).Should(Equal(0), fmt.Sprintf("failed to remove all pod objects from namespace %s", namespace))
+
+				err = cleanNamespaceLabels(namespace)
+			}
+
+			err := changeManagerReplicas(defaultNumberOfReplicas)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create a pod when mac pool is running in a regular namespace", func() {
+			err := setRange(rangeStart, rangeEnd)
+			Expect(err).ToNot(HaveOccurred())
+
+			podObject := CreatePodObject(nil)
+
+			Eventually(func() bool {
+				_, err := testClient.KubeClient.CoreV1().Pods(TestNamespace).Create(podObject)
+				if err != nil && strings.Contains(err.Error(), "connection refused") {
+					return false
+				}
+
+				return true
+			}, timeout, pollingInterval).Should(BeTrue(), "failed to apply the new pod object")
+		})
+
+		It("should fail to create a pod on a regular namespace when mac pool is down", func() {
+			err := changeManagerReplicas(0)
 			Expect(err).ToNot(HaveOccurred())
+
+			podObject := CreatePodObject(nil)
+
+			Eventually(func() error {
+				_, err := testClient.KubeClient.CoreV1().Pods(TestNamespace).Create(podObject)
+				return err
+
+			}, timeout, pollingInterval).Should(HaveOccurred(), "failed to apply the new pod object")
+		})
+
+		It("should create a pod on a critical k8s namespaces when mac pool is down", func() {
+			err := changeManagerReplicas(0)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = addLabelsToNamespace(OtherTestNamespace, map[string]string{names.K8S_RUNLABEL: "0"})
+			Expect(err).ToNot(HaveOccurred())
+
+			podObject := CreatePodObject(nil)
+
+			Eventually(func() bool {
+				_, err := testClient.KubeClient.CoreV1().Pods(OtherTestNamespace).Create(podObject)
+				if err != nil && strings.Contains(err.Error(), "connection refused") {
+					return false
+				}
+
+				return true
+			}, timeout, pollingInterval).Should(BeTrue(), "failed to apply the new pod object")
+		})
+
+		It("should create a pod on a critical openshift namespaces when mac pool is down", func() {
```

Comment:
Could you please use an inline helper function `testCriticalNamespace := func ...` for this one and the Kubernetes one? Maybe even the valid one can be handled via the helper function.

Reply:
done